src/features/island/collectibles/Collectible.tsx

```diff
@@ -353,16 +356,34 @@ const InProgressCollectible: React.FC<Props> = ({
   );
 };
 
+const selectCrops = (state: MachineState) => state.context.state.crops;
+const selectCollectibles = (state: MachineState) =>
+  state.context.state.collectibles;
+
+const compareCrops = () => true;
+const compareCollectibles = () => true;
+
 const CollectibleComponent: React.FC<Props> = ({
   name,
   id,
   readyAt,
   createdAt,
-  coordinates,
+  x,
+  y,
   isRustyShovelSelected,
   showTimers,
-  grid,
 }) => {
+  const { gameService } = useContext(Context);
+
+  const crops = useSelector(gameService, selectCrops, compareCrops);
+  const collectibles = useSelector(
+    gameService,
+    selectCollectibles,
+    compareCollectibles
+  );
+
+  const gameGrid = getGameGrid({ crops, collectibles });
```

Comment:
@Polysemouse `getGameGrid` is a bit CPU intensive. Initially it was in `Land.tsx` so it only ran once each re-render, now it runs in each `Collectible.tsx` which could cause some issues.

I would suggest leaving it in `Land.tsx` for now and we can abstract it into the gameMachine context at a later date to update only when positions change on the map

Reply:
addressed